subprojects/core-api/src/main/java/org/gradle/api/tasks/Classpath.java

```diff
@@ -28,6 +28,13 @@
  * <p>This annotation should be attached to the getter method in Java or the property in Groovy.
  * Annotations on setters or just the field in Java are ignored.</p>
  *
+ * <p>
+ *     For jar files, the normalized path is empty.
+ *     The content of the jar file is normalized so that time stamps and order of the zip entries in the jar file do not matter.
+ *     If a directory is a classpath entry, then the directory itself is ignored.
```

Comment:
We should be clear that this is only about the root directory.

Reply:
I changed it, hoping to make it clear.